src/main/java/org/project/springweb/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+package org.project.springweb.service.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.project.springweb.dto.order.CreateOrderRequestDto;
+import org.project.springweb.dto.order.OrderResponseDto;
+import org.project.springweb.dto.order.UpdateOrderStatusDto;
+import org.project.springweb.dto.orderitem.OrderItemResponseDto;
+import org.project.springweb.exception.EntityNotFoundException;
+import org.project.springweb.exception.OrderProcessingException;
+import org.project.springweb.mapper.OrderItemMapper;
+import org.project.springweb.mapper.OrderMapper;
+import org.project.springweb.model.Order;
+import org.project.springweb.model.OrderItem;
+import org.project.springweb.model.ShoppingCart;
+import org.project.springweb.model.Status;
+import org.project.springweb.repository.order.OrderRepository;
+import org.project.springweb.repository.orderitem.OrderItemRepository;
+import org.project.springweb.repository.shoppingcart.ShoppingCartRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    public OrderResponseDto createOrder(CreateOrderRequestDto requestDto, Long userId) {
```

Comment:
Should you clear the shopping cart after completing the order? Also, let’s use Transactional ;)